org.lflang/src/org/lflang/generator/EclipseErrorReporter.java

```diff
@@ -109,21 +111,12 @@ private String report(String message, Severity severity, Integer line, Path file
         if (line == null || file == null)
             System.err.println(header + ": " + message);
         else
-            System.err.println(header + ": " + file + " line " + line
-                    + "\n" + message);
-
-        // Determine the iResource to report on
-        IResource iResource = file != null ? FileUtil.getIResource(file) : null;
-        // if we couldn't find an iResource (for whatever reason), then use the
-        // iResource of the main file
-        if (iResource == null) {
-            iResource = fileConfig.iResource;
-        }
+            System.err.println(header + ": " + file + " line " + line + "\n" + message);
 
         // Create a marker in the IDE for the error.
         // See: https://help.eclipse.org/2020-03/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Fguide%2FresAdv_markers.html
         try {
-            IMarker marker = iResource.createMarker(IMarker.PROBLEM);
+            IMarker marker = bestEffortGetIResource(file).createMarker(IMarker.PROBLEM);
```

Comment:
I al ready mentioned this, but for the record: this solution should be factored into the existing `FileUtil.getIResource` method.

Reply:
Ah, sorry, I forgot to push.